common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -169,8 +169,8 @@ trait Logging {
           }
           Logging.sparkShellThresholdLevel = replLevel
           rootLogger.getAppenders().asScala.foreach {
-            case (_, ca: ConsoleAppender) =>
-              ca.addFilter(new SparkShellLoggingFilter())
+            case (_, appender: AbstractAppender) =>
+              appender.addFilter(new SparkShellLoggingFilter())
```

Comment:
Doesn't this add the filter to all appenders of the root logger? I think the original intention here is to only add the filter to console only. I think by doing this, the logging of other appenders will be affected.

Reply:
+1, for console only

Reply:
+1

Reply:
> Doesn't this add the filter to all appenders of the root logger? I think the original intention here is to only add the filter to console only. I think by doing this, the logging of other appenders will be affected.

AsyncAppender -> ConsoleAppender then can't work....hmmm